plugins/ring/src/main.ts

```diff
@@ -690,6 +690,18 @@ export class RingLocationDevice extends ScryptedDeviceBase implements DeviceProv
             }
         }
         location.onLocationMode.subscribe(updateLocationMode);
+        
+        // if the location has a base station, updates when arming/disarming are not sent to the `onLocationMode` subscription
+        // instead we subscribe to the security panel, which is updated during arming actions
+        location.getSecurityPanel().then(panel => {
+            panel.onData.subscribe(_ => { 
+                location.getLocationMode().then(response => {
+                    updateLocationMode(response.mode);
+                });
+            });
+        }).catch(_ => {
+            this.console.warn(`no security panel found, updates will not be recieved on this subscription`)
```

Comment:
is there a hasSecurityPanel you can check before invoking this? users will get weirded out by debug messages that look like errors.

Reply:
There is not, I will just not log the error and put a comment in the catch statement